### PR TITLE
remove deprecated kwarg "context"

### LINKED ISSUE #773: Backport a fix for pyepics context kwarg warning
This warning

```
/srv/conda/envs/notebook/lib/python3.7/site-packages/epics/pv.py:110: UserWarning: The `context` kwarg for epics.get_pv() is deprecated. New PVs will _not_ be created in the requested context.
```

mentioned also by @prjemian in #765 should be back-ported to 1.3.x since we still have some work to do before releasing 1.4.0.

### PR DESCRIPTION
**Pull request by jklynch**
This PR resolves #773.

### CODE REVIEW

**Comment by tacaswell** [comment]
I am a bit concerned about this as I think this is required to make ophyd work with older pyepics and I'm not sure we want to force that forward here.

**Comment by jklynch** [comment]
On Nov 5 I asked @prjemian and @ZLLentz if dropping support for python 3.5 from ophyd 1.3.x would affect them and they said they have no python 3.5 deployments.

**Comment by jklynch** [comment]
@tacaswell I guess you were talking about the deprecated keyword arg

**Comment by jklynch** [comment]
There is some evidence this PR is not compatible with pyepics 3.3.0, but it is easy to upgrade to 3.4.0.

**Comment by danielballan** [comment]
@jklynch If we want ophyd to _work_ with old pyeipcs (satisfying @tacaswell's concern above) but avoid issuing a warning with new pyepics (resolving #773) then we could remove the pin on `pyepics` and instead do a check to see which version of pyepics we have to decide whether to use the `context` kwarg or not.

```py
from distutils.version import LooseVersion    
LooseVersion(epics.__version__) >= LooseVersion('3.4.0')
```

File: ophyd/_pyepics_shim.py [hunk 1]
```diff
@@ -19,8 +20,8 @@
 
 
 def get_pv(*args, **kwargs):
-    import epics
-    kwargs.setdefault('context', epics.ca.current_context())
+    if not LooseVersion(epics.__version__) >= LooseVersion('3.4.0'):
```
Comment (danielballan) [inline ophyd/_pyepics_shim.py]: Minor readability improvement by avoiding the double-negative.

```suggestion
    if LooseVersion(epics.__version__) < LooseVersion('3.4.0'):
```

Comment (danielballan) [reply ophyd/_pyepics_shim.py]: Might be worth linking, in a comment, to this commit where the change we are accounting for here was introduced. https://github.com/pyepics/pyepics/commit/1ca46afa5afe285fecd4ea1d12a7eeee0376c773

h/t @tacaswell for digging that up.

File: .travis.yml [hunk 2]
```diff
@@ -22,8 +22,6 @@ cache:
 
 matrix:
   include:
-    - python: 3.5
```
Comment (tacaswell) [inline .travis.yml]: do we have to drop 3.5 testing?  Although I initially suggested we do that, I thought we settled on removing the one f-string in a backport.

**Comment by tacaswell** [comment]
pyepics 3.4 is the right version to check, came in via https://github.com/pyepics/pyepics/commit/1ca46afa5afe285fecd4ea1d12a7eeee0376c773

File: .travis.yml (hunk 2 above)
Comment (danielballan) [reply .travis.yml]: It's not about language features we're using; it's about packages available from conda. (Note that this is a `CONDA_ENV=1` build.) However, now that we have (rightly!) removed the pyepics pin that was once in this PR, perhaps we can restore the 3.5 build and remove the `python_requires`.

Comment (danielballan) [reply .travis.yml]: To underline the point: thanks for the close read, @tacaswell!

Comment (tacaswell) [reply .travis.yml]: fair cop.

**Pull request merged**